[Wang Hong Yong] iP

asherhy opened the pull request:
# Duke (NotPro Version)
> “Your mind is for having ideas, not holding them.” –David Allen ([Source](https://dansilvestre.com/productivity-quotes/))

Duke frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2122S1/ip/pull/365).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 

**And it is FREE!**  :smile:

Features:

- [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

asherhy committed:
Level-1

asherhy committed:
commit 2

asherhy committed:
commit 3

asherhy committed:
commit 4

asherhy committed:
A-TextUiTesting

asherhy committed:
commit 5

asherhy committed:
commit 6

asherhy committed:
Update Duke.java

asherhy committed:
Add DukeStorage.java and TaskHandler.java

asherhy committed:
Update Duke.java, Deadline.java, Event.java, Task.java, Todo.java

asherhy committed:
Update Deadline.java

asherhy committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java

asherhy committed:
Add Command classes

asherhy committed:
Add Ui class

asherhy committed:
Update Storage

asherhy committed:
Update TaskList

asherhy committed:
Update Duke.java

asherhy committed:
Update Deadline.java

asherhy committed:
Update Task.java

asherhy committed:
Create Parser.java

asherhy committed:
Add Packages

asherhy committed:
Add JUnit Tests

asherhy committed:
Create MANIFEST.MF

asherhy committed:
Add JavaDoc

asherhy committed:
update CodingStandard

asherhy committed:
Update CodingStandard

asherhy committed:
Add Level-9

asherhy committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/parser/Parser.java

asherhy committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/ui/Ui.java

asherhy committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Bluntsord commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,118 @@
+        String[] info = s.split(" \\| ");
+        Task task;
+
+        switch (info[0]) {
```
Please name the variables instead of accessing array each time

Bluntsord commented on src/main/java/duke/data/task/Deadline.java:
```diff
@@ -0,0 +1,80 @@
+        try {
+            if (dateTime.split("\\s+").length == 2) {
+                sourceFormat = DateTimeFormatter.ofPattern("uuuu-MM-dd HHmm");
+                targetFormat = DateTimeFormatter.ofPattern("MMM dd uuuu hh:mm a", Locale.ENGLISH);
```
Nice. Pretty clean if I do say so myself

Bluntsord commented on src/main/java/duke/data/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+     * @return string representation of the done status of the task.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
Its been a while since i've seen ternary in java. Nice

Bluntsord commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,77 @@
+        switch (commandInput) {
+        case "list":
+            command = new ListCommand(taskList);
+            command.execute();
```
Move command.execute to the bottom? This ways it prevents repetition

Bluntsord reviewed (commented):
Really clean code. Didn't find any functions that were too long. Nice application of slap. Mistakes pointed out are minor and do not change much. Keep up the good work!

asherhy committed:
Add Gradle

asherhy committed:
Add Checkstyle

frederickpek commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,139 @@
+
+    private static final String DIVIDER = "____________________________________________________________";
+    private static final String TAB = "  ";
+    private static final String LS = System.lineSeparator();
```
You might want to give LS a more defined name and fully spell it out in caps, since it has a large scope and is a constant as well.

frederickpek commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,123 @@
+ */
+public class Storage {
+    private static final String FILE_NAME = "./data/allTasks.txt";
+    private static final File file = new File(FILE_NAME);
```
Same here, for 'file'.

frederickpek commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,151 @@
+    public void findTask(String wordToFind) {
+        ArrayList<String> tasks = new ArrayList<>();
+        int index = 1;
+        for (Task t: list) {
```
I like that the spaces between operators are consistent throughout your code, but you missed one over here!

frederickpek reviewed (commented):
I like the cleanliness of your code! Just some minor coding standards violations to clean up 👍🏻

asherhy committed:
Update allTasks.txt

asherhy committed:
Update Level-10

asherhy committed:
Update Javadoc

asherhy committed:
Add Assertions

Certain assumptions is made and must be fulfilled in order for the program to run successfully.

Assertions are used to define assumptions about the program state so that the runtime can verify them.

Assertion stops the program from running and the sooner it stops, the safer it is.

Commit:
Improve Code Quality

There are some parts of the code not following the standard conventions from textbook.

Added some important parts and removed some unnecessary comments.

Let's refactor the code to be more readable and understandable.

asherhy committed:
Merge pull request #1 from asherhy/branch-A-Assertions

Add Assertions

asherhy committed:
Merge branch 'master' into branch-A-CodeQuality

asherhy committed:
Merge pull request #2 from asherhy/branch-A-CodeQuality

Branch a code quality

asherhy committed:
Add C-Tagging

asherhy committed:
Fix bugs

asherhy committed:
Add Error handling and java doc

asherhy committed:
Update README.md

asherhy committed:
Add Ui.png

asherhy committed:
Update DukeApp.java

asherhy committed:
Update UI

asherhy committed:
Update MainWindow.java

asherhy committed:
Update README.md

asherhy committed:
Update MainWindow.java
